zend_abstract_interface/json/json.c

```diff
@@ -22,7 +22,7 @@ __attribute__((weak)) int php_json_encode(smart_str *buf, zval *val, int options
 __attribute__((weak)) int php_json_decode_ex(zval *return_value, const char *str, size_t str_len, zend_long options,
                                              zend_long depth);
 #endif
-__attribute__((weak)) zend_class_entry *php_json_serializable_ce;
+extern __attribute__((weak)) zend_class_entry *php_json_serializable_ce;
```

Comment:
Hmm. Using `extern` and `weak` on the same symbol doesn't make sense to me. What does that even mean?

Reply:
I am not 100% sure, but the following is happening on MacOS if `ext-json` is statically linked:
`php_json_serializable_ce` from `ext/json/json.c` will be initialised in the JSON extensions `MINIT`, but in our `zai_json_setup_bindings()` it is still `NULL`. Adding `extern` to it makes it show me the same pointer.

Reply:
Macos actually requires extern weak... it doesn't make sense, but macos wants that.

Reply:
Just for future reference the Link Bob posted in Slack
https://developer.apple.com/library/archive/documentation/MacOSX/Conceptual/BPFrameworks/Concepts/WeakLinking.html